src/filesystem/impls/filer/ArchiveUtils.js

```diff
@@ -16,6 +16,46 @@ define(function (require, exports, module) {
     var Path           = Filer.Path;
     var fs             = Filer.fs();
 
+    // Mac and Windows clutter zip files with extra files/folders we don't need
+    function _skipFile(filename) {
+        var basename = Path.basename(filename);
+
+        // Skip OS X additions we don't care about in the browser fs
+        if(/^__MACOSX\//.test(filename)) {
+            // http://superuser.com/questions/104500/what-is-macosx-folder
+            return true;
+        }
+        if(basename === ".DS_Store") {
+            // https://en.wikipedia.org/wiki/.DS_Store
+            return true;
+        }
+        if(/^\._/.test(basename)) {
+            // http://apple.stackexchange.com/questions/14980/why-are-dot-underscore-files-created-and-how-can-i-avoid-them
+            return true;
+        }
+
+        // Skip Windows additiosn we don't care about in the browser fs
+        if(/(Tt)humbs\.db/.test(basename)) {
+            // https://en.wikipedia.org/wiki/Windows_thumbnail_cache
+            return true;
+        }
+        if(basename === "desktop.ini") {
+            // http://www.computerhope.com/issues/ch001060.htm
+            return true;
+        }
+
+        // Include this file, don't skip.
+        return false;
+    }
+
+    function _refreshFilesystem(callback) {
+        // Update the file tree to show the new files
+        CommandManager.execute(Commands.FILE_REFRESH).always(function() {
+            // Generate Blob URLs for all the files we imported
+            BlobUtils.preload(StartupState.project("root"), callback);
```

Comment:
What `FILE_REFRESH` does is tell the Brackets file cache to reload from disk and update the filetree so it shows the new file(s)/folder(s).  However, this won't create Blob URLs for any of them, so if extract an image, it won't really be there.  This forces everything to reload into Blob URLs.  For all existing files, that's a no-op.  For new stuff, it will get cached so things work as you expect.